### PR TITLE
fix: [#176660226] Fix iPhone 12 (Mini, Pro...) home button overlapping onto the tab bar

### PR DESCRIPTION
**Pull request by DavideValdo**
## List of changes proposed in this pull request
Before: 
before

After:
after

## How to test
Open the app with any iPhone of the iPhone 12 product line, see the tab bar not overlapping.

### CODE REVIEW

File: ts/navigation/MainNavigator.tsx [hunk 1]
```diff
@@ -44,16 +45,34 @@ const getIcon = (routeName: string): string => {
...
     paddingRight: 3,
     borderTopWidth: 0,
-    paddingTop: 8,
-    paddingBottom: 10
-  },
+    paddingTop: 8
+  };
+
+  // Add space to the bottom bar for iPhone 12
+  const isIPhone12 = deviceInfoModule.getDeviceId().indexOf("iPhone13") !== -1;
```
Comment (CrisTofani) [inline ts/navigation/MainNavigator.tsx:58]:
```suggestion
  const isIPhone12 = deviceInfoModule.getDeviceId().indexOf("iPhone12") !== -1;
```

Did you mean `iPhone12`?

**Review by CrisTofani — CHANGES_REQUESTED** [review]

File: ts/navigation/MainNavigator.tsx (hunk 1 above)
Comment (DavideValdo) [reply ts/navigation/MainNavigator.tsx:61]: No, the `deviceId`s for the iPhone 12 product line are "13,1"; "13,2" ... See https://gist.github.com/adamawolf/3048717, also confirmed through `console.log`. Added comment for this.

File: ts/components/__tests__/FiscalCodeLandscapeOverlay.test.tsx [hunk 2]
```diff
@@ -17,6 +17,15 @@ import { PreferredLanguageEnum } from ".../../../definitions/backend/PreferredLa
 
 jest.mock("react-native-share", () => jest.fn());
 
+jest.mock("react-native-device-info", () => {
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply ts/components/__tests__/FiscalCodeLandscapeOverlay.test.tsx:20]: Obv this comment is valid for all the mocked tests

**Review by CrisTofani — CHANGES_REQUESTED** [review]

**Review by Undermaken — APPROVED** [review]

File: ts/navigation/MainNavigator.tsx [hunk 3]
```diff
@@ -44,16 +45,37 @@ const getIcon = (routeName: string): string => {
...
+  // See https://gist.github.com/adamawolf/3048717 for device IDs
+
+  const isIPhone12 = deviceInfoModule.getDeviceId().indexOf("iPhone13") !== -1;
+
+  return {
+    ...defaultStyle,
+    ...(isIPhone12
+      ? {
+          paddingBottom: 20,
+          height: 74
```
Comment (CrisTofani) [inline ts/navigation/MainNavigator.tsx:68]: Trying this solution I just notice some minimal differences between iPhone 11 and 12 space between home button and navigator bar



i just tried out with 
```typescript
        {
          paddingBottom: 30,
          height: 84
        }
```

and seems to be closer as style what do you think?

**Review by CrisTofani — CHANGES_REQUESTED** [review]

File: ts/navigation/MainNavigator.tsx (hunk 3 above)
Comment (DavideValdo) [reply ts/navigation/MainNavigator.tsx:68]: Yes! Applying this change.

**Review by CrisTofani — APPROVED** [review]

**Pull request merged**